packages/fiori-app-sub-generator/src/fiori-app-generator/end.ts

```diff
@@ -141,14 +141,16 @@ export async function runPostGenerationTasks(
     // Persist backend system connection information
     const hostEnv = getHostEnvironment();
 
-    if (service.backendSystem && hostEnv !== hostEnvironment.bas && service.backendSystem.newOrUpdated) {
+    if (service.backendSystem && hostEnv !== hostEnvironment.bas && service.backendSystem.newOrUpdated === true) {
         const storeService = await getService<BackendSystem, BackendSystemKey>({
             logger: logger,
             entityName: 'system'
         });
         // No need to await, we cannot recover anyway
         // eslint-disable-next-line @typescript-eslint/no-floating-promises
         storeService.write(service.backendSystem, { force: true });
+    } else if (service.backendSystem?.newOrUpdated === false && hostEnv !== hostEnvironment.bas) {
```

Comment:
This condition is unnecessary. you cant determine if the user declined here anyway